### PR TITLE
Add SRIOV Operator CI lane

### PR DESCRIPTION
**Pull request by zeeke**
Add a Github Action lane to trigger the SRIOV Network Operator tests. This allows to verify if PR changes break any working scenario.

### CODE REVIEW

File: .github/workflows/buildtest.yml [hunk 1]
```diff
@@ -56,3 +56,56 @@ jobs:
...
+        uses: docker/setup-buildx-action@v2
+
+      - name: Login to Docker
+        uses: docker/login-action@v2
+        with:
+          registry: ghcr.io
+          username: ${{ github.repository_owner }}
+          password: ${{ secrets.GITHUB_TOKEN }}
+
+      - name: Build and push sriov-cni
```
Comment (adrianchiris) [inline .github/workflows/buildtest.yml:77]: this is going to accumulate allot of "junk" 

cant we build the image locally and use it ? as part of virtual-k8s-cluster step ?

Comment (zeeke) [reply .github/workflows/buildtest.yml:77]: Good point. To use locally built images, we have to modify the scripts in the operator and push passed images to the k8s internal registry [run-e2e-conformance-virtual-cluster.sh#L226](https://github.com/k8snetworkplumbingwg/sriov-network-operator/blob/master/hack/run-e2e-conformance-virtual-cluster.sh#L226)

Another way to overcome this is to use a cleaner like [GHCR Delete Image](https://github.com/marketplace/actions/ghcr-delete-image), though I'm not sure how mature that project is

Comment (e0ne) [reply .github/workflows/buildtest.yml:77]: can we re-use image published by https://github.com/k8snetworkplumbingwg/sriov-cni/blob/master/.github/workflows/image-push-master.yml?

Comment (zeeke) [reply .github/workflows/buildtest.yml:77]: > can we re-use image published by https://github.com/k8snetworkplumbingwg/sriov-cni/blob/master/.github/workflows/image-push-master.yml?

`image-push-master.yml` publishes the image after PR is merged. With these changes, I'd like to run end-to-end tests against proposed changes. So, we need a way to build and publish (maybe in a temporary registry) an image different than the master one.

**Comment by zeeke** [comment]
/hold
waiting for
- https://github.com/k8snetworkplumbingwg/sriov-network-operator/pull/518

**Comment by zeeke** [comment]
This is now ready for review, it worked against sriov-cni wip PR:
- https://github.com/k8snetworkplumbingwg/sriov-cni/pull/282

File: .github/workflows/buildtest.yml [hunk 2]
```diff
@@ -56,3 +56,42 @@ jobs:
...
+      - name: Check out the repo
+        uses: actions/checkout@v3
+
+      - name: build sriov-cni image
+        run: podman build -t ghaction-sriov-cni:pr-${{github.event.pull_request.number}} .
+  
+      - name: Check out sriov operator's code
+        uses: actions/checkout@v2
+        with:
+          repository: zeeke/sriov-network-operator-1
```
Comment (mlguerrero12) [inline .github/workflows/buildtest.yml:76]: zeeke?

**Comment by zeeke** [comment]
Sorry, I put the wrong PR on review. This PR is still **on hold** because it waits 
- https://github.com/k8snetworkplumbingwg/sriov-network-operator/pull/518

to be merged

File: .github/workflows/buildtest.yml [hunk 3]
```diff
@@ -56,3 +56,42 @@ jobs:
...
+          ref: e2e-inject-images
+          path: sriov-network-operator-wc
+
+      - name: fix sriov-device-plugin
```
Comment (SchSeba) [inline .github/workflows/buildtest.yml:80]: why do we need this? the image used by the operator for the device plugin is not good?

Comment (zeeke) [reply .github/workflows/buildtest.yml:80]: We don't need this. I'll remove it as soon as the operator's PR is merged:
- https://github.com/k8snetworkplumbingwg/sriov-network-operator/pull/518

**Comment by zeeke** [comment]
@SchSeba @adrianchiris Please take a look

**Review by SchSeba — APPROVED** [review]
LGTM

**Pull request merged**